File: CountPairs.java
int countPairs(int arr[], int target) {
        // Your code here
        Map<Integer, Integer> map = new HashMap<>();
        int cnt = 0;
        for(int num : arr){
            int compliment = target - num;
            if(map.containsKey(compliment)){
                cnt += map.get(compliment);
            }
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        return cnt;
    }

File: CountPairs.cpp
int countPairs(vector<int> &arr, int target) {
        // Code here
        unordered_map<int, int> map;
        int cnt = 0;
        
        for(int num : arr) {
            int compliment = target - num;
            if(map.find(compliment) != map.end()) {
                cnt += map[compliment];
            }
            
            map[num]++;
        }
        
        return cnt;
    }
